processor/src/main/java/io/jonasg/bob/BuilderTypeSpecFactory.java

```diff
@@ -116,37 +116,58 @@ private List<MethodSpec> generateSetters() {
 	}
 
 	protected MethodSpec generateSetterForField(BuildableField field) {
+
 		var builder = MethodSpec.methodBuilder(setterName(field.fieldName()))
 				.addModifiers(Modifier.PUBLIC)
 				.returns(builderType())
 				.addParameter(TypeName.get(field.type()), field.fieldName());
-		if (field.isConstructorArgument() && isEnforcedConstructorPolicy() || field.isMandatory()) {
+		if (field.isConstructorArgument() && isAnEnforcedConstructorPolicy() || field.isMandatory()) {
 			builder.addStatement("this.$L.set($L)", field.fieldName(), field.fieldName());
+			if (isEnforcedAllowNullsConstructorPolicy()) {
```

Comment:
This should be done inside the `RequiredField` class in doing so **encapsulating** both setting the `boolean` and the actual value.